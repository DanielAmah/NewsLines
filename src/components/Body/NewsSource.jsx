import React from 'react';
import PropTypes from 'prop-types';

/**
 * @class Sources
 * @extends {React.Component}
 * @param {string}
 * @returns {sources} a div with source title and sources description
 */
class NewsSource extends React.Component {
  handleQueryValue(href) {
    this.context.router.push(href);
  }
  render() {
    const { name, description, id, sortBysAvailable } = this.props;
    const link = `/news?name=${name}&id=${id}&sorts=${sortBysAvailable}`;
    return (
      <div className="col-md-3" id="source">
        <center><h3 className="sourceTitle">{name}</h3></center>
        <center><p>{description.slice(0, 50)}...</p></center>
        <center>
        <button
          className="btn btn-danger"
          onClick={this.handleQueryValue.bind(this, link)}
        >
          View Headlines
        </button>
        </center>
      </div>
    );
  }
}
/**
 * Default props
 */
NewsSource.defaultProps = {
  name: '',
  description: '',
  id: '',
  sortBysAvailable: []
};

/**
 * Default props 
 */
NewsSource.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  id: PropTypes.string,
  sortBysAvailable: PropTypes.array
};

/**
 * Set Props
 */
NewsSource.contextTypes = {
  router: PropTypes.object
};

export default NewsSource;
