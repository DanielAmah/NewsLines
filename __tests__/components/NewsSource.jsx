/* global expect jest test */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import NewsSource from '../../src/components/Body/NewsSource';

describe('Sources Component', () => {
  test('should match the sources snapshot', () => {
    const component = mount(<NewsSource />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('should contain the appropraite props', () => {
    const wrapper = mount(<NewsSource />);
    expect(wrapper.node.props.name).toEqual('');
    wrapper.setProps({ name: 'the-next-web' });
    expect(wrapper.node.props.name).toEqual('the-next-web');
  });

  test('props should be in correct positions', () => {
    const wrapper = mount(<NewsSource name="Daniel Amah" description="Software Developer" />);
    const h3 = wrapper.find('h3');
    const p = wrapper.find('p');
    expect(h3.text()).toBe('Daniel Amah');
    expect(p.text()).toBe('Software Developer...');
  });
});
