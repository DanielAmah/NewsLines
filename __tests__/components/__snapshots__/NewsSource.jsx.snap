// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources Component should match the sources snapshot 1`] = `
<NewsSource
  description=""
  id=""
  name=""
  sortBysAvailable={Array []}
>
  <div
    className="col-md-3"
    id="source"
  >
    <center>
      <h3
        className="sourceTitle"
      />
    </center>
    <center>
      <p>
        ...
      </p>
    </center>
    <center>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        View Headlines
      </button>
    </center>
  </div>
</NewsSource>
`;
